### PR TITLE
fix: change getAsset to not return a promise

### PR DESCRIPTION
**Pull request by erezrokah**
Initial proposed approach:

1. Return a new instance of the bounded `getAsset` to force re-render.
2. Make sure relevant `shouldComponentUpdate` methods check for changes in `getAsset`.
3. Use `isLoadingAsset` to trigger another render once the asset is loaded.

**Also**
Moved `boundGetAsset` from `Editor.js` to `EditorPreviewPane.js` (Similar  to `EditorControl.js`).
 
**Todo**
- [x] Check compatibility with editor components.

### CODE REVIEW

File: packages/netlify-cms-widget-file/src/withFileControl.js [hunk 1]
```diff
@@ -118,9 +108,9 @@ export default function withFileControl({ forImage } = {}) {
 
     shouldComponentUpdate(nextProps) {
       /**
-       * Always update if the value changes.
+       * Always update if the value or getAsset changes.
        */
-      if (this.props.value !== nextProps.value) {
+      if (this.props.value !== nextProps.value || this.props.getAsset !== nextProps.getAsset) {
```
Comment (barthc) [inline packages/netlify-cms-widget-file/src/withFileControl.js]: Some performance issues here, `getAsset` is a function and between props updates, the above check will always pass. Editing any other widget will always force this widget to re-render.

Comment (erezrokah) [reply packages/netlify-cms-widget-file/src/withFileControl.js]: Actually though about that and we could memoize it here https://github.com/netlify/netlify-cms/blob/8d67de0e681e02ddbf811eecc5e4744b324b7e61/packages/netlify-cms-core/src/actions/media.ts#L85 if we it causes issue

**Comment by marcveens** [comment]
Just a question from my point of view, this is a breaking change but is released in a minor version update of netlify-cms. Should this maybe be treated differently, like in a major release?

**Comment by erquhart** [comment]
We broke this by making it asynchronous in a recent non-major release, this is more of a fix/revert of that. We also have a problem with semver adherence in packages that rely on other packages within the monorepo, that’s the real issue.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**